Issue #819: FileContentsHolder does not have HTML documentation

MEZk opened the pull request:
@romani 
>1) please create xdoc documentation for FileContentsHolder

Done.

>2) update UT that check that all Checks are present to validate also all filters and other modules of Checker and TreeWalker.

Done.

> 3) We need to split page http://checkstyle.sourceforge.net/config.html into few pages (filter should go to separate sub page for sure). Please propose new structure.

Splitted page http://checkstyle.sourceforge.net/config.html into 11 pages. Each page describe one aspect of checkstyle's configuration as a result there is no mess in documentation. 

Site after changes:
http://mezk.github.io/site/

romani commented:
Good step forward , but too much separate pages:

1)
http://mezk.github.io/site/config_filters.html#Overwiev

fix typo - overview.

2)
> http://mezk.github.io/site/config_xmldetails.html

XML structure

3)
> http://mezk.github.io/site/config_overview.html

move this content to http://mezk.github.io/site/config.html

4)
> http://mezk.github.io/site/config_properties.html
http://mezk.github.io/site/property_types.html

combine these pages together. Each Type should be a section as we often need to share a link to exact type when we discuss something in issue and mail-list (I understand that this not ideal view, but we need direct links).

5)
> http://mezk.github.io/site/config_modules.html
http://mezk.github.io/site/config_checker.html
http://mezk.github.io/site/config_treewalker.html
http://mezk.github.io/site/config_auditlisteners.html 

please combine them in one page. Only http://mezk.github.io/site/config_filters.html should stay separate as it contain big content and used a lot.

6)
I plan to make a release during late this Sunday, it would so great to release your changes as 6.11.2

MEZk commented:
@romani 
1) Done. Moved brief description of properties to config.html. 
2) Done.
3) Done.
4) Done. Restored location of properties section in config.html. Made each type as a section in property_types.html.
5) Separated only filters page. Added link to config_filters.html in Filters section of config.html.

http://mezk.github.io/site/

romani commented:
merged as FF
